src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py

```diff
@@ -98,7 +98,6 @@ def create(self, size, isolate_nspace, pool, mode, uid, gid):
             create_base_dir(self.fs, self.group.path, self.vol_spec.DEFAULT_MODE)
             # create directory and set attributes
             self.fs.mkdirs(subvol_path, mode)
-            self.mark_subvolume()
```

Comment:
Why does this fail? This should just set an xattr on the UUID dir (for v1 subvolume).

Reply:
This is because by the time `self.mark_subvolume()` is called, the `"path"` meta item hasn't been set for the subvolume. That gets set below at [line 113]( https://github.com/ceph/ceph/blob/e57773464537beef07eb6aa98034b6eb73a80d74/src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py#L113)

Reply:
If you look at [these lines](https://github.com/ceph/ceph/blob/e57773464537beef07eb6aa98034b6eb73a80d74/src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py#L54-L75), you can see that `self.mark_subvolume()` tries to set the `ceph.dir.subvolume` xattr on `self.path`

If you look at the `path` property of v1 subvolume, it tries to fetch it from the `MetadataMgr` object.

Reply:
Thanks for explaining. That indeed is a bug, its just that we don't create v1 subvolumes anymore, so this code path does get any tests :(

We should include the fix however :)